packages/react-flex/Spec.md

```diff
@@ -0,0 +1,524 @@
+| `direction`       | string = "`row`", "`row-reverse`", "`column`", "`column-reverse`"                                     | "`row`"       | To further cement the idea of Flex being an abstraction of flexbox, this would bring the user to a closer experience of flexbox while also being less biased on our interpretation of the default direction. Absorbs Stack's `reversed`. |
+| `horizontalAlign` | string (_see [align-items](https://developer.mozilla.org/en-US/docs/Web/CSS/justify-content#values)_) | "`normal`"    | This name gives the user a more verbose and easily recognizable idea for what the prop does. Absorbs Flex's `space`.                                                                                                                     |
+| `verticalAlign`   | string (_see [justify-content](https://developer.mozilla.org/en-US/docs/Web/CSS/align-items#values)_) | "`normal`"    | _see `horizontalAlign`'s comments._                                                                                                                                                                                                      |
+| `gap`             | string (_see [margin](https://developer.mozilla.org/en-US/docs/Web/CSS/margin#values)_)               | `0`           | `margin` setter for each of the children items.                                                                                                                                                                                          |
+| `wrap`            | boolean                                                                                               | `false`       | Interfaces are already aligned. Simplification of `flex-wrap`.                                                                                                                                                                           |
+| `as`              | `React.ElementType<React.HTMLAttributes<HTMLElement>>`                                                | "`div`"       | Perserving same implementation.                                                                                                                                                                                                          |
+| `grow`            | boolean \| number \| string = "`inherit`", "`initial`", "`unset`"                                     | `0`           | Focusing again on abstracting, `grow` will affect all the FlexItem's `flex-grow` styles. A `true` value will translate into `1` for ease of use and retro-compatibility with Stack.                                                      |
```

Comment:
**thought:** cant we just drop the `boolean` ? I understand from compatibility reasons it might be easier to leave as is, but from DX perspective it feels confusing. Also to support it additional logic would need to be present in the implementation I guess.

Reply:
I'm kind of torn between these. Simply abstracting `flex-grow` (and `flex-shrink`) and allowing exclusively the same values as that takes seems like the better option, but I personally like `<Component prop />` kind of usage. However, I'm definitely with you on keeping our implementation simpler at the cost of just 1 more breaking change.

Reply:
yeah, I mean having boolean props is okish whenever it makes sense (we can take DOM as a source of good inspiration/familiarity and  among developers) -> based on that, in this case, it doesn't make sense

Reply:
Let's go with this then, makes usage closer to `flex-grow` and simplifies the implementation.